libkbfs/ops.go

```diff
@@ -134,6 +178,17 @@ func (oc *OpCommon) getFinalPath() path {
 	return oc.finalPath
 }
 
+func (oc *OpCommon) checkUpdatesValid() error {
```

Comment:
Instead of this, you could have a global function that takes an `op` and just iterates over `op.AllUpdates()`.  That way you wouldn't need a separate `checkValid()` on each op type that has to remember to check the custom types.